### PR TITLE
porting dynamic-sidecar to pyndatic2

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

Currently I'm stuck, waiting for service-library service to be properly fixed.

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/dynamic-sidecar/openapi.json [hunk 1]
```diff
@@ -213,7 +213,15 @@
             "content": {
               "application/json": {
                 "schema": {
-                  "$ref": "#/components/schemas/ActivityInfo"
+                  "anyOf": [
```
Comment (giancarloromeo) [inline services/dynamic-sidecar/openapi.json:216]: Did you notice that now (it seems OpenAPI 3.1) the `nullable` fields have this new schema with `anyOf` and the new type `"null"`?

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/settings.py [hunk 2]
```diff
@@ -155,7 +161,7 @@ class ApplicationSettings(BaseCustomSettings, MixinLoggingSettings):
     RABBIT_SETTINGS: RabbitSettings = Field(auto_default_from_env=True)
 
     DY_DEPLOYMENT_REGISTRY_SETTINGS: RegistrySettings = Field()
-    DY_DOCKER_HUB_REGISTRY_SETTINGS: RegistrySettings | None = Field()
+    DY_DOCKER_HUB_REGISTRY_SETTINGS: RegistrySettings | None = Field(default=None)
 
     RESOURCE_TRACKING: ResourceTrackingSettings = Field(auto_default_from_env=True)
```
Comment (giancarloromeo) [inline services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/settings.py]: Use of `Field(auto_default_from_env=True)` as extra is deprecated and will be removed in Pydantic v3.

Comment (giancarloromeo) [inline services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/settings.py]:
```suggestion
    RESOURCE_TRACKING: ResourceTrackingSettings = Field(json_schema_extra={"auto_default_from_env": True})
```

**Review by giancarloromeo — COMMENTED** [review]
Thanks a lot! Just a couple of comments from my side.

**Comment by giancarloromeo** [comment]
We should bring this one direct to `pydantic_v2_migration`.

**Comment by GitHK** [comment]
closing in favour of https://github.com/ITISFoundation/osparc-simcore/pull/6534

**Pull request closed without merging**